presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -112,4 +122,33 @@ public static boolean isEqualComparisonExpression(Expression expression)
     {
         return expression instanceof ComparisonExpression && ((ComparisonExpression) expression).getOperator() == ComparisonExpression.Operator.EQUAL;
     }
+
+    public static Optional<EnumType> maybeGetEnumTypeFromLiteral(DereferenceExpression node, TypeManager typeManager)
+    {
+        try {
+            Type baseType = typeManager.getType(parseTypeSignature(node.getBase().toString()));
+            if (baseType instanceof EnumType) {
+                return Optional.of((EnumType) baseType);
+            }
+        }
+        catch (IllegalArgumentException e) {
+            return Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    public static boolean isEnumLiteral(DereferenceExpression node, Type nodeType)
+    {
+        return nodeType instanceof EnumType && node.getBase().toString().equalsIgnoreCase(nodeType.toString());
+    }
+
+    public static Object resolveEnumLiteral(DereferenceExpression literal, EnumType enumType)
```

Comment:
Either put validation that the node is a valid enum, or make sure only valid enum can be used to call this. I think this function should call `isEnumLiteral`.